### DISCUSSION TITLE
Markdown Automatic URL Linking Not Working

### DISCUSSION THREAD

**Discussion by AgedLace** [mkdocs-publisher/mkdocs-publisher#38] [General]
I have created two new files - basic-markdown.md and -extended-markdown.md based upon the tutorials from [The Markdown Guide Site](https://www.markdownguide.org/).

Everything works perfectly except for the [Automatic URL Linking.](https://www.markdownguide.org/extended-syntax/#automatic-url-linking)  I'm guessing this is not something that has anything to do with your plugins, but maybe a result of Material for MkDocs??

I am attaching the logs, but I'm not seeing any warnings or errors in my terminal.

I will start going through Obsidian's Syntax examples tomorrow.  I hope these things are helpful to you.

[20231003_184231_mkdocs_build.log](https://github.com/mkusz/mkdocs-publisher/files/12795956/20231003_184231_mkdocs_build.log)
[20231003_184231_mkdocs_debug.zip](https://github.com/mkusz/mkdocs-publisher/files/12795957/20231003_184231_mkdocs_debug.zip)

**Comment by mkusz** [comment]
Hi,
The web page you have mentioned has one crucial sentence:

> Many Markdown processors automatically turn URLs into links.

"Many" means not all and the Markdown processor used by MkDocs doesn't support this out of the box.
As for Material for MkDocs, it allows for some additional stuff, but it's a skin/theme for MkDocs and mostly rely on MkDocs Markdown processor (with some exceptions).

So is there a way to fix it?

Yes ;)

Before we can start, you should fully understand how Markdown processor in MkDocs can be configured. Probably you are able to get a great answer by pointing me to the `mkdocs.yml` file. Inside this file, there is a section called `markdown_extensions` with a list of extensions being used. Material for MkDocs and MkDocs by itself are heavily relied on that list. In the `mkdocs.yml` file in this project repo, I have added most of the existing extensions because simply I use them on the project documentation. I skipped one that is need fot mentioned functionality. The missing extension is called `pymdownx.magiclink`. To make it work, you have to add it to the `markdown_extensions` section.

Thanks for that, and I will need to add this to the template repo when I find time to finally make it ;)

PS. There are 2 lists where additional extensions can be found:
- https://python-markdown.github.io/extensions/
- https://facelessuser.github.io/pymdown-extensions/extensions/arithmatex/

**Reply by AgedLace** [reply]
I am somewhat familiar with Material MkDocs extensions - I have read their docs several times over the course of several projects.  My problem is that I don't use the majority of those extensions because they were beyond my ability to understand them.  The more I work with you on this project, the more things are beginning to make sense and I'm learning to experiment somewhat with new things.  

Usually when I create a project using Material for MkDocs, I generally enable all of their pymdownx extensions, even if I don't understand them, or believe I will ever use them.  The *only* extension I didn't enable was this very one - because it seems to deal with links, I thought it *might* interfere with the wikilinks you are working on in this project. I have enabled it and you were right, this extension fixed this problem.  

You can close this discussion as 'solved'!

Thank you for the extension links for more research  ... I will review them both - I have been learning a great deal since working with you on this project, and I am very grateful for your patience and help!!!

Thank you!!!

**Comment by mkusz** [comment]
@AgedLace The cooperation with you also is giving me some additional motivation to work on that project and shows me some directions when I can improve it or fix it.
